edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -60,43 +60,62 @@ def get_jwt_claim_mergeable_attributes(self):
 
     def authenticate(self, request):
+        is_jwt_from_authorization_header = self._is_jwt_from_authorization_header(request)
+
         try:
             user_and_auth = super().authenticate(request)
 
-            # Unauthenticated, CSRF validation not required
+            # Unable to authenticate, proceed to next Authentication class
             if not user_and_auth:
-                return user_and_auth
+                return None
 
-            # Not using JWT cookies, CSRF validation not required
-            use_jwt_cookie_requested = request.META.get(USE_JWT_COOKIE_HEADER)
-            if not use_jwt_cookie_requested:
+            # JWT found in authorization header, CSRF validation not required
+            if is_jwt_from_authorization_header:
                 return user_and_auth
 
+            # TODO: Why do we need CSRF protection but the original class does not?
+            # Enforce CSRF for JWT Cookie (assumed if not using the authorization header)
             self.enforce_csrf(request)
 
             # CSRF passed validation with authenticated user
             return user_and_auth
 
-        except jwt.InvalidTokenError as token_error:
-            # Note: I think this case is not used, but will monitor the custom attribute to verify.
-            set_custom_attribute('jwt_auth_failed', 'InvalidTokenError:{}'.format(repr(token_error)))
-            raise exceptions.AuthenticationFailed() from token_error
-        except Exception as exception:
-            # Errors in production do not need to be logged (as they may be noisy),
-            # but debug logging can help quickly resolve issues during development.
-            logger.debug('Failed JWT Authentication,', exc_info=exception)
-            # Note: I think this case should only include AuthenticationFailed and PermissionDenied,
-            #   but will monitor the custom attribute to verify.
-            set_custom_attribute('jwt_auth_failed', 'Exception:{}'.format(repr(exception)))
-            raise
+        except (AuthenticationFailed, PermissionDenied) as exception:
+            if is_jwt_from_authorization_header:
+                logger.debug('Invalid JWT auth header.', exc_info=exception)
+                set_custom_attribute('jwt_auth_failed', 'auth header:{}'.format(repr(exception)))
+                raise
+
+            # TODO: Add test coverage
+            # If we reach here, we must be using the JWT Cookie.
+            # For failed authentication using JWT Cookie, we will allow other Authentication classes
+            # to have a chance to attempt authentication by returning none, but will log and add monitoring.
+            logger.debug('Invalid JWT Cookie. Proceeding to next Authentication class.', exc_info=exception)
+            set_custom_attribute('jwt_auth_failed', 'cookie:{}'.format(repr(exception)))
+            return None
+
+    @classmethod
+    def _is_jwt_from_authorization_header(cls, request):
```

Comment:
Reviewer Note: I don't love this method of determining when JWT cookies were used. It seems prone to error, but I can't think of a great way to determine this based on the super class implementation.  One alternative is to [override get_token_from_cookies](https://github.com/Styria-Digital/django-rest-framework-jwt/blob/88096b16d8639336fcb34d1e3420fafd1dc7c8f7/src/rest_framework_jwt/authentication.py#L122) and capture if it was called for the request (possibly in the request cache) and if so, assume we've moved on to cookies.